lib/util/http/rate-limits.ts

```diff
@@ -0,0 +1,87 @@
+import is from '@sindresorhus/is';
+import { matchesHost } from '../host-rules';
+import * as hostRules from '../host-rules';
+import type { RateLimitRule } from './types';
+
+const defaults: RateLimitRule[] = [
+  {
+    // https://guides.rubygems.org/rubygems-org-rate-limits/
+    matchHost: 'rubygems.org',
+    throttleMs: 125,
+  },
+  {
+    // https://crates.io/data-access#api
+    matchHost: 'https://crates.io/api/',
+    throttleMs: 1000,
+  },
+  {
+    matchHost: '*',
+    concurrency: 16,
+  },
+];
+
+let limits: RateLimitRule[] = [];
+
+export function setHttpRateLimits(rules: RateLimitRule[] = defaults): void {
+  limits = rules;
+}
```

Comment:
this is not fail save against runtime null passing

Reply:
In the runtime, it's meant to be called with no arguments, hence set `defaults`.
For tests, it remains empty everywhere except `rate-limit.spec.ts`.

What would you like to change?

Reply:
```suggestion
export function setHttpRateLimits(rules?: RateLimitRule[]): void {
  limits = rules ?? defaults;
```
so it's null and undefined safe for any invalid code ussge